### PR TITLE
Add tbls format

### PR DESCRIPTION
**Pull request by arthaud-proust**
See the associated issue :
- #471 

Code was generated by Claude, reviewed and tested by a human

### CODE REVIEW

**Comment by kerrizor** [comment]
I've taken an initial look here and the implementation looks solid -- good test coverage, handles edge cases like polymorphic associations, and follows the existing generator patterns.

**One thing I'd like to discuss before merging:**

In your issue you mentioned an alternative approach:

> "Instead, we can imagine a json export that would not be tied to any specific use nor format, allowing erd-renderers to branch themselves to rails-erd."

I think this is worth considering more seriously. A **generic JSON export** of rails-erd's domain model could:

1. Support TBLS/Liam ERD (your use case)
2. Support future tools without new generators
3. Reduce coupling to external format specs
4. Let the community build adapters externally

The question is: **is the TBLS format close enough to "generic" that it serves this purpose?** Looking at the schema, it's fairly standard (tables, columns, constraints, relations) — not overly TBLS-specific.

A few options I see:

| Option | Pros | Cons |
|--------|------|------|
| **A) Merge as-is** | Works now, well-tested, solves your problem | Tied to TBLS format |
| **B) Rename to `json` generator** | More generic naming, same output | Still TBLS-shaped, might confuse users expecting "generic" |
| **C) True generic JSON + external TBLS adapter** | Maximum flexibility | More work, delays your use case |

I'm leaning toward **Option A** (merge as-is) since:
- TBLS is a reasonable, well-documented format
- Both tbls and Liam are healthy projects (4k+ stars each)
- You've already done the work and it's solid
- We can always add a more generic export later if demand emerges

File: test/unit/tbls_test.rb [hunk 1]
```diff
@@ -0,0 +1,150 @@
+require File.expand_path("../test_helper", File.dirname(__FILE__))
```
Comment (kerrizor) [inline test/unit/tbls_test.rb:3]:
```suggestion
# frozen_string_literal: true

require File.expand_path("../test_helper", File.dirname(__FILE__))
```

**Comment by kerrizor** [comment]
@arthaud-proust A couple of small things before we merge:

1. **Add `frozen_string_literal: true`** to the top of `test/unit/tbls_test.rb` (we use it consistently across the codebase)

2. **Add a CHANGES.md entry** under the Unreleased section:
   ```markdown
   ### New Features
   * Add tbls JSON generator for integration with Liam ERD and other tbls-compatible tools (#471, #474)
   ```

One thing I noticed while reviewing: the tbls schema also supports a top-level relations array in addition to the FK constraints on tables. Your current implementation embeds the FK info as constraints, which works perfectly for Liam ERD and is totally valid.

Some tbls-consuming tools might expect that relations array too, but honestly -- let's not over-engineer this. What you have works, it's well-tested, and it solves the problem you set out to solve. If someone runs into a tool that needs the relations array, that's a great follow-up PR for them (or us) to tackle.

Ship it, iterate later! 🚀

Let me know when you've pushed those two small fixes and I'll get this merged.

File: lib/rails_erd/cli.rb [hunk 2]
```diff
@@ -193,10 +193,10 @@ def start
...
       @path, @options = path, options
       generator = options[:generator] || RailsERD.options[:generator]
-      if generator == :mermaid
-        require "rails_erd/diagram/mermaid"
-      else
-        require "rails_erd/diagram/graphviz"
+      case generator
+      when :mermaid then require "rails_erd/diagram/mermaid"
+      when :tbls    then require "rails_erd/diagram/tbls"
+      else               require "rails_erd/diagram/graphviz"
```
Comment (kerrizor) [inline lib/rails_erd/cli.rb]: The current `else` clause makes it look like Graphviz is the default, when actually `:mermaid` is the default (set in `RailsERD.default_options`). Consider making this more explicit:

```suggestion
    def initialize(path, options)
      @path, @options = path, options
      generator = options[:generator] || RailsERD.options[:generator]
      case generator
      when :mermaid  then require "rails_erd/diagram/mermaid"
      when :tbls     then require "rails_erd/diagram/tbls"
      when :graphviz then require "rails_erd/diagram/graphviz"
      else                require "rails_erd/diagram/mermaid"  # default
      end
    end
```

Same for the `generator` method (lines 237-244):

```suggestion
    def generator
      generator_type = options[:generator] || RailsERD.options[:generator]
      case generator_type
      when :mermaid  then RailsERD::Diagram::Mermaid
      when :tbls     then RailsERD::Diagram::Tbls
      when :graphviz then RailsERD::Diagram::Graphviz
      else                RailsERD::Diagram::Mermaid  # default
      end
    end
```

This way the code documents its own behavior -- if someone passes an invalid generator name, they get the default (mermaid) rather than silently falling back to graphviz.

**Comment by kerrizor** [comment]
### CI Failure Analysis 🔍

The failures are all from one test — the rest of the output is just warnings from dependencies (frozen string literals, sqlite fork safety, etc.).

**The actual failure:**

```
TblsTest#test_columns_carry_name,_type,_nullable;_default_and_comment_only_when_set [test/unit/tbls_test.rb:61]:
Expected: "varchar"
  Actual: "varchar(456)"
```

Different Rails/SQLite versions report `sql_type` differently — some return `varchar`, others return `varchar(456)` with the length specifier. The assertion is too strict.

**Suggested fix** in `test/unit/tbls_test.rb` line 61:

```ruby
# Instead of:
assert_equal "varchar", name_col[:type]

# Use:
assert name_col[:type].start_with?("varchar"), "Expected varchar type, got #{name_col[:type]}"
```

This makes the test resilient across Rails/SQLite versions while still validating the behavior.

File: lib/rails_erd/cli.rb (hunk 2 above)
Comment (arthaud-proust) [reply lib/rails_erd/cli.rb]: Ok, changed. I'll document this behavior change in the changelog

**Comment by arthaud-proust** [comment]
Thanks for the review, I handled you request changes and suggestions

**Comment by kerrizor** [comment]
@arthaud-proust thanks again! I'm going to merge this and roll it into a new release, thank you!
